bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -0,0 +1,954 @@
+            Address::Ed25519(ed) => AddressDto::Ed25519(ed.into()),
+            _ => unimplemented!(),
+        }
+    }
+}
+
+impl TryFrom<&AddressDto> for Address {
+    type Error = Error;
+
+    fn try_from(value: &AddressDto) -> Result<Self, Self::Error> {
+        match value {
+            AddressDto::Ed25519(a) => Ok(Address::Ed25519(a.try_into()?)),
+        }
+    }
+}
+
+/// Describes an Ed25519 address.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Ed25519AddressDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    pub address: String,
+}
+
+impl From<&Ed25519Address> for Ed25519AddressDto {
+    fn from(value: &Ed25519Address) -> Self {
+        Self {
+            kind: Ed25519Address::KIND,
+            address: value.to_string(),
+        }
+    }
+}
+
+impl TryFrom<&Ed25519AddressDto> for Ed25519Address {
+    type Error = Error;
+
+    fn try_from(value: &Ed25519AddressDto) -> Result<Self, Self::Error> {
+        value
+            .address
+            .parse::<Ed25519Address>()
+            .map_err(|_| Error::InvalidSyntaxField("address"))
+    }
+}
+
+/// Describes a treasury output.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TreasuryOutputDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    pub amount: u64,
+}
+
+/// Describes all the different unlock types.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum UnlockBlockDto {
+    Signature(SignatureUnlockDto),
+    Reference(ReferenceUnlockDto),
+}
+
+impl From<&UnlockBlock> for UnlockBlockDto {
+    fn from(value: &UnlockBlock) -> Self {
+        match value {
+            UnlockBlock::Signature(s) => match s {
+                SignatureUnlock::Ed25519(ed) => UnlockBlockDto::Signature(SignatureUnlockDto {
+                    kind: SignatureUnlock::KIND,
+                    signature: SignatureDto::Ed25519(Ed25519SignatureDto {
+                        kind: Ed25519Signature::KIND,
+                        public_key: hex::encode(ed.public_key()),
+                        signature: hex::encode(ed.signature()),
+                    }),
+                }),
+                _ => unimplemented!(),
+            },
+            UnlockBlock::Reference(r) => UnlockBlockDto::Reference(ReferenceUnlockDto {
+                kind: ReferenceUnlock::KIND,
+                index: r.index(),
+            }),
+            _ => unimplemented!(),
+        }
+    }
+}
+
+impl TryFrom<&UnlockBlockDto> for UnlockBlock {
+    type Error = Error;
+
+    fn try_from(value: &UnlockBlockDto) -> Result<Self, Self::Error> {
+        match value {
+            UnlockBlockDto::Signature(s) => match &s.signature {
+                SignatureDto::Ed25519(ed) => {
+                    let mut public_key = [0u8; ED25519_ADDRESS_LENGTH];
+                    hex::decode_to_slice(&ed.public_key, &mut public_key)
+                        .map_err(|_| Error::InvalidSyntaxField("publicKey"))?;
+                    // TODO const
```

Comment:
What does this `TODO` mean?

Reply:
There is a hardcoded value that needs to be replaced by an existing constant